vulnfeeds/cmd/cperepos/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"context"
+	"encoding/xml"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"sort"
+
+	"cloud.google.com/go/logging"
+	"github.com/google/osv/vulnfeeds/cves"
+	"github.com/google/osv/vulnfeeds/utility"
+	"golang.org/x/exp/slices"
+)
+
+type CPEDict struct {
+	XMLName  xml.Name  `xml:"cpe-list"`
+	CPEItems []CPEItem `xml:"cpe-item"`
+}
+
+type CPEItem struct {
+	XMLName    xml.Name    `xml:"cpe-item" json:"-"`
+	Name       string      `xml:"name,attr" json:"name"`
+	Title      string      `xml:"title" json:"title"`
+	References []Reference `xml:"references>reference" json:"references"`
+	CPE23      CPE23Item   `xml:"cpe23-item" json:"cpe23-item"`
+}
+
+type Reference struct {
+	URL         string `xml:"href,attr" json:"URL"`
+	Description string `xml:",chardata" json:"description"`
+}
+
+type CPE23Item struct {
+	Name string `xml:"name,attr"`
+}
+
+const (
+	CPEDictionaryDefault = "cve_jsons/nvdcpematch-1.0.json"
+	projectId            = "oss-vdb"
+)
+
+var (
+	Logger            utility.LoggerWrapper
+	ValidDescriptions = []string{
+		"Version",
+		"Vendor",
+		"Change Log",
+		"Product",
+		// Brings in https://github.com/CVEProject/cvelist, which is not relevant
+		// "Advisory",
+		"Project",
+		"Vendor website", // unlikely to be a repo, but statistically significant as a description
+		"product changelog",
+		"Version information",
+		"vendor website", // unlikely to be a repo, but statistically significant as a description
+		"Vendor Website", // unlikely to be a repo, but statistically significant as a description
+		"version information",
+		"product version information",
+		"product information",
+		"Product changelog",
+		"vendor product information",
+		"Changelog",
+		"Version Information",
+		"Vendor changelog",
+		"project information",
+		"product release information",
+		"product release notes",
+		"vendor product website",
+		"Product version information",
+		"vendor changelog",
+	}
+	// These repos should never be considered authoritative for a product.
+	InvalidRepos = []string{
+		"https://github.com/CVEProject/cvelist", // Heavily in Advisory URLs, sometimes shows up elsewhere
+		"https://github.com/github/cvelist",     // Fork of the above
+	}
+	CPEDictionaryFile = flag.String("cpe_dictionary", CPEDictionaryDefault, "CPE Dictionary file to parse")
+)
+
+func LoadCPEDictionary(f string) (CPEDict, error) {
+	xmlFile, err := os.Open(f)
+	if err != nil {
+		Logger.Fatalf("Failed to open %s: %v", f, err)
+	}
+
+	defer xmlFile.Close()
+
+	byteValue, _ := ioutil.ReadAll(xmlFile)
+
+	var c CPEDict
+	xml.Unmarshal(byteValue, &c)
+
+	return c, nil
+}
+
+func main() {
```

Comment:
Maybe add a line or two of comments explaining what the output of this is. It wasn't clear that the immediate output of this is the description frequency analysis, as opposed to the CPE -> repo mappings themselves.

Reply:
I will ditch this entirely in favour of emitting a CSV file in addition to the JSON file in the followup PR I'll make.